src/gas-price/gas-price.ts

```diff
@@ -0,0 +1,205 @@
+import { ethers } from 'ethers';
+
+import type { GasSettings } from '../config/schema';
+
+interface DataFeedValue {
+  value: ethers.BigNumber;
+  timestampMs: number;
+}
+interface GasState {
+  gasPrices: { price: ethers.BigNumber; timestampMs: number }[];
+  lastOnChainDataFeedValues: Record<string, DataFeedValue>;
+}
+
+export const gasPriceStore: Record<string, Record<string, GasState>> = {};
+
+export const initializeGasStore = (chainId: string, providerName: string) => {
```

Comment:
I think we should have just a single state, so I'd move merge this logic with `src/state`.

Reply:
Done, that was quite a refactor 😄